### PR TITLE
NT-0045 - Admin crud + refactor userController/middlewares

### PR DESCRIPTION
**Pull request by kamilmoskal**

### CODE REVIEW

File: server/middleware/auth.ts [hunk 1]
```diff
@@ -17,12 +18,21 @@ export const auth = (
...
-		if (withEmail) {
+		if (authEmail) {
 			if (!user.emailVerified) {
 				return res.status(401).json({ msg: messages.EMAIL_NOT_VERIFIED });
 			}
 		}
-		req.user = user;
+		if (authAdmin){
+			if(!user.roles.includes('admin')){
+				return res.status(401).json({ msg: "You don't have permission to access" });
```
Comment (mras2303) [inline server/middleware/auth.ts]: add this text to variable

File: server/middleware/auth.ts [hunk 2]
```diff
@@ -17,12 +18,21 @@ export const auth = (
...
 		}
-		if (withEmail) {
+		if (authEmail) {
 			if (!user.emailVerified) {
 				return res.status(401).json({ msg: messages.EMAIL_NOT_VERIFIED });
 			}
 		}
-		req.user = user;
+		if (authAdmin){
+			if(!user.roles.includes('admin')){
```
Comment (mras2303) [inline server/middleware/auth.ts]: move this string to variable

File: server/routes/api/admin.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import express, { NextFunction, Request, Response } from 'express';
+
```
Comment (mras2303) [inline server/routes/api/admin.ts]: Remove empty line

File: server/routes/api/admin.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { auth } from '../../middleware/auth';
+
```
Comment (mras2303) [inline server/routes/api/admin.ts]: Remove empty line

File: server/routes/api/admin.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import * as usersController from '../../controllers/users/users';
+
+const router = express.Router();
+
+// @route   GET api/admin/userList
+// @desc    Return user list
+// @access  Private
+router.get(
+    '/userList',
+    (req: Request, res: Response, next: NextFunction) => auth(req, res, next, true, true),
```
Comment (mras2303) [inline server/routes/api/admin.ts]: what does it mean true, true?

Comment (kamilmoskal) [reply server/routes/api/admin.ts]: @mras2303 
4th argument means that route including authencitation for 'emailVerified' field,
5th argument including authencitation for admin

Comment (mras2303) [reply server/routes/api/admin.ts]: is there any possibility to write it in better way?

Comment (kamilmoskal) [reply server/routes/api/admin.ts]: You tell me :D I gonna try find out

Comment (kamilmoskal) [reply server/routes/api/admin.ts]: I read that if we're using more than 1 optional parameter, we should use object so we can do smth like:
**in auth.ts:**
`
export const auth = (
	req: Request,
	res: Response,
	next: NextFunction,
	{authEmail = true, authAdmin = false} : {authEmail?: boolean, authAdmin?: boolean },
) => {....
`
**and then in route without parameters:**
`(req: Request, res: Response, next: NextFunction) => auth(req, res, next, {}),`
**with 1 any optional parameter:**
`(req: Request, res: Response, next: NextFunction) => auth(req, res, next, {authAdmin: true}),`

Comment (mras2303) [reply server/routes/api/admin.ts]: To further discuss

Comment (Foukz) [reply server/routes/api/admin.ts]: I think object with settings params is great option

Comment (kamilmoskal) [reply server/routes/api/admin.ts]: i've changed this and now it's pretty short and clean

**Pull request merged**